pkg/controller/migration/namespace_migration.go

```diff
@@ -571,23 +573,33 @@ func (m *CoreNamespaceMigration) getNodesToMigrate() []*v1.Node {
 
 // waitUntilNodeCanBeMigrated checks the number of desired and ready pods in the kube-system and calico-system
 // daemonsets to make sure we don't simultaneously migrate more pods than allowed.
-func (m *CoreNamespaceMigration) waitUntilNodeCanBeMigrated(ctx context.Context) error {
+func (m *CoreNamespaceMigration) waitUntilNodeCanBeMigrated(ctx context.Context, log logr.Logger) error {
 	return wait.PollImmediate(1*time.Second, 1*time.Minute, func() (bool, error) {
+		// num node desired schedule, num node ready, node max unavailable in kube-system
 		ksD, ksR, _, err := m.getNumPodsDesiredAndReady(ctx, kubeSystem, nodeDaemonSetName)
 		if err != nil {
 			return false, err
 		}
+
+		// num node desired schedule, num node ready, node max unavailable in calico-system
 		csD, csR, csMaxUnavailable, err := m.getNumPodsDesiredAndReady(ctx, common.CalicoNamespace, nodeDaemonSetName)
 		if err != nil {
 			return false, err
 		}
 
-		var maxUnavailable int32 = 1
+		var maxUnavailable int32 = int32(defaultMaxUnavailable)
```

Comment:
can we set this to using int ? or should we set the windows one to using int32 ?

Reply:
Any API fields should not use `int` and should instead use either `int32` or `int64` per the API conventions, so eventually this needs to be converted to an in32 (since that's the type of the maxunavailable field in the API)

> All public integer fields MUST use the Go (u)int32 or Go (u)int64 types, not (u)int (which is ambiguous depending on target platform). Internal types may use (u)int.

https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/api-conventions.md

Reply:
changing the windows on to int32 https://github.com/tigera/operator/pull/1804/commits/6882da130e88f4cba96d1f1738e44ac45cb41414